ddtrace/contrib/internal/ray/utils.py

```diff
@@ -0,0 +1,219 @@
+# to the job name train_my_model
+JOB_NAME_REGEX = re.compile(r"^job\:([A-Za-z0-9_\.\-]+),run:([A-Za-z0-9_\.\-]+)$")
+# The entry point regex is intended to extract the name of the Python script from a Ray entrypoint,
+# for example, if the entrypoint is python3 woof.py --breed mutt
+# then the job name will be woof
+ENTRY_POINT_REGEX = re.compile(r"([^\s\/\\]+)\.py")
+
+
+def _inject_dd_trace_ctx_kwarg(method: Callable) -> Signature:
+    old_sig = inspect.signature(method)
+    if DD_TRACE_CTX in old_sig.parameters:
+        return old_sig
+
+    new_param = Parameter(DD_TRACE_CTX, Parameter.KEYWORD_ONLY, default=None)
+    params_list = list(old_sig.parameters.values()) + [new_param]
+    sorted_params = sorted(params_list, key=lambda p: p.kind == Parameter.VAR_KEYWORD)
+    return old_sig.replace(parameters=sorted_params)
+
+
+def _inject_context_in_kwargs(context, kwargs):
+    headers = {}
+    _TraceContext._inject(context, headers)
+    if "kwargs" not in kwargs or kwargs["kwargs"] is None:
+        kwargs["kwargs"] = {}
+    kwargs["kwargs"][DD_TRACE_CTX] = headers
+
+
+def _inject_context_in_env(context):
+    headers = {}
+    _TraceContext._inject(context, headers)
+    os.environ["traceparent"] = headers.get("traceparent", "")
+    os.environ["tracestate"] = headers.get("tracestate", "")
+
+
+def _extract_tracing_context_from_env():
+    if os.environ.get("traceparent") is not None and os.environ.get("tracestate") is not None:
+        return _TraceContext._extract(
+            {
+                "traceparent": os.environ.get("traceparent"),
+                "tracestate": os.environ.get("tracestate"),
+            }
+        )
+    return None
+
+
+def _inject_ray_span_tags_and_metrics(span):
+    span.set_tag_str("component", "ray")
+    span.set_metric(_DJM_ENABLED_KEY, 1)
+    span.set_metric(_FILTER_KEPT_KEY, 1)
+    span.set_metric(_SPAN_MEASURED_KEY, 1)
+    span.set_metric(_SAMPLING_PRIORITY_KEY, 2)
+
+    submission_id = os.environ.get(RAY_SUBMISSION_ID)
+    if submission_id is not None:
+        span.set_tag_str(RAY_SUBMISSION_ID_TAG, submission_id)
+
+    if ray.is_initialized():
+        runtime_context = get_runtime_context()
+
+        span.set_tag_str(RAY_JOB_ID, runtime_context.get_job_id())
+        span.set_tag_str(RAY_NODE_ID, runtime_context.get_node_id())
+
+        worker_id = runtime_context.get_worker_id()
+        if worker_id is not None:
+            span.set_tag_str(RAY_WORKER_ID, worker_id)
+
+        if runtime_context.worker.mode == ray._private.worker.WORKER_MODE:
+            task_id = runtime_context.get_task_id()
+            if task_id is not None:
+                span.set_tag_str(RAY_TASK_ID, task_id)
+
+        actor_id = runtime_context.get_actor_id()
+        if actor_id is not None:
+            span.set_tag_str(RAY_ACTOR_ID, actor_id)
+
+
+def set_maybe_big_tag(span, tag_name, tag_value):
```

Comment:
```suggestion
def set_tag_if_safe(span, tag_name, tag_value):
```
maybe?
I am not a huge fan of the original name, and it could be a little more descriptive

Reply:
addressed in [936e2f8](https://github.com/DataDog/dd-trace-py/pull/14511/commits/936e2f8879995831232f579c7b92f200c53d0e01)